### PR TITLE
Unit Tests for Messages Reactions and Replies

### PR DESCRIPTION
**Pull request by erinb90**
Issues #115, #147

### CODE REVIEW

**Review by g-harel — DISMISSED** [review]

File: test/unitTest/java/org/thoughtcrime/securesms/messagereactions/ReactionsHandlerTests.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        assertEquals(results.get(2).getReaction(), ":)");
+
+        assertEquals(results.get(0).getReactor().toString(), "reactor id");
+        assertEquals(results.get(0).getReactionDate(), (Long) 1234L);
+        assertEquals(results.get(0).getReactionId(), "reaction id");
+        assertEquals(results.get(0).getMessageHash(), "message hash");
+    }
+
+    @Test
+    public void testGetReactionCounts() {
```
Comment (veken1199) [inline test/unitTest/java/org/thoughtcrime/securesms/messagereactions/ReactionsHandlerTests.java]: could you test when there are no reactions ??? @erinb90 The case where a message does not have any reactions

File: test/unitTest/java/org/thoughtcrime/securesms/messagereactions/ReactionsHandlerTests.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    public void testGetMessageReactions() {
+        List<ReactionsHandler.Reaction> results = reactionsHandler.getMessageReactions(messageRecord);
+
+        assertEquals(results.size(), 3);
+
+        assertEquals(results.get(0).getReaction(), ":)");
+        assertEquals(results.get(1).getReaction(), ":(");
+        assertEquals(results.get(2).getReaction(), ":)");
+
+        assertEquals(results.get(0).getReactor().toString(), "reactor id");
```
Comment (veken1199) [inline test/unitTest/java/org/thoughtcrime/securesms/messagereactions/ReactionsHandlerTests.java:53]: I like this assert

File: test/unitTest/java/org/thoughtcrime/securesms/messagereplies/RepliesHandlerTests.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    @Test
+    public void testReceiveReply() {
+        repliesHandler.receiveReply(messageHash, replies[0], time, replierId, threadId);
+        verify(messageReplyDatabase).replyToMessage(messageHash, replies[0], time, replierId, threadId);
+    }
+
+    @Test
+    public void testGetMessageReplies() {
```
Comment (veken1199) [inline test/unitTest/java/org/thoughtcrime/securesms/messagereplies/RepliesHandlerTests.java:43]: Again, Can you test the condition when a message has 0 replies ??

File: test/unitTest/java/org/thoughtcrime/securesms/messagereactions/ReactionsHandlerTests.java (hunk 1 above)
Comment (erinb90) [reply test/unitTest/java/org/thoughtcrime/securesms/messagereactions/ReactionsHandlerTests.java]: will do

**Review by veken1199 — APPROVED** [review]
@erinb90 submitted the changes I requested. the pr looks good

**Comment by g-harel** [comment]
@veken1199 why was this pr closed?

**Comment by erinb90** [comment]
@g-harel I think that was a mistake

**Review by shelbaz — APPROVED** [review]
Looks good

**Pull request merged**